pythran/pythonic/numpy/nanmax.hpp

```diff
@@ -15,27 +15,38 @@ namespace numpy
   namespace
   {
     template <class E, class F>
-    void _nanmax(E begin, E end, F &max, utils::int_<1>)
+    bool _nanmax(E begin, E end, F &max, utils::int_<1>)
     {
+      bool found = false;
       for (; begin != end; ++begin) {
         auto curr = *begin;
-        if (!functor::isnan()(curr) && curr > max)
+        if (!functor::isnan()(curr) && curr >= max) {
           max = curr;
+          found = true;
+        }
       }
+      return found;
     }
+
     template <class E, class F, size_t N>
-    void _nanmax(E begin, E end, F &max, utils::int_<N>)
+    bool _nanmax(E begin, E end, F &max, utils::int_<N>)
     {
+      bool found = false;
       for (; begin != end; ++begin)
-        _nanmax((*begin).begin(), (*begin).end(), max, utils::int_<N - 1>());
+        found |= _nanmax((*begin).begin(), (*begin).end(), max,
+                         utils::int_<N - 1>());
+      return found;
     }
   }
 
   template <class E>
   typename E::dtype nanmax(E const &expr)
   {
+    bool found = false;
     typename E::dtype max = std::numeric_limits<typename E::dtype>::lowest();
-    _nanmax(expr.begin(), expr.end(), max, utils::int_<E::value>());
+    found = _nanmax(expr.begin(), expr.end(), max, utils::int_<E::value>());
+    if (!found)
```

Comment:
Let me propose an alternative implementation: set `max` to `nan`, do not track `found` and return `max` in the end. For that trick to work, one need to change the test

```
if (!functor::isnan()(curr) && curr > max)
```
to

```
if (!functor::isnan()(curr) && !(curr <= max))
```

it makes the code cleaner, and probably faster, based on two limit cases:
- https://quick-bench.com/q/zJrmiX0h3SsJERLQ9l97lHLMy-g (no NaN)
- https://quick-bench.com/q/AYPj5AXwjBT_kUxgs43ZgK-XvLw (all NaN)

Reply:
Oh and by the way: thanks for working on this, and for providing test cases, clang-formating the code etc. that's much appreciated ;-)

Reply:
ok for me. I thought it was costly to compare with nan but it seems it is not the case any more

Reply:
I was surprised too :-)